client/asset/btc/btc.go

```diff
@@ -4347,6 +4347,104 @@ func (btc *baseWallet) createWitnessSig(tx *wire.MsgTx, idx int, pkScript []byte
 	return sig, privKey.PubKey().SerializeCompressed(), nil
 }
 
+// EstimateSendTxFee returns a tx fee estimate for sending or withdrawing the
+// provided amount using the provided feeRate.
+func (btc *baseWallet) EstimateSendTxFee(address string, sendAmount, feeRate uint64, subtract bool) (fee uint64, err error) {
+	if sendAmount == 0 {
+		return 0, fmt.Errorf("cannot check fee: send amount = 0")
+	}
+
+	_, err = btc.decodeAddr(address, btc.chainParams)
+	if err != nil {
+		return 0, fmt.Errorf("invalid address: %s", address)
+	}
+
+	feeRate = btc.feeRateWithFallback(feeRate)
+
+	minTxSize := uint64(dexbtc.MinimumTxOverhead + btc.outputSize())
```

Comment:
The output address determines the output size.  `btc.outputSize` is for when we know the output address will be segwit or not based on the wallet config, but you can send/withdraw to any address.  It's just a mater of bytes, but I think this is where you should use the address.  Just validating the address but nothing else seems out of scope for this method.